pkg/storage/streams/store.go

```diff
@@ -84,22 +104,78 @@ func (s *streamStore) Put(ctx context.Context, path paths.Path, data io.Reader,
 	var totalSize int64
 	var lastSegmentSize int64
 
-	awareLimitReader := EOFAwareReader(data)
+	var startingNonce [24]byte
+	_, err = rand.Read(startingNonce[:])
+	if err != nil {
+		return Meta{}, err
+	}
+	// copy startingNonce so that startingNonce is not modified by the encrypter before it is saved to lastSegmentMeta
+	var nonce [24]byte
+	copy(nonce[:], startingNonce[:])
+
+	derivedKey, err := path.DeriveContentKey(s.rootKey)
+	if err != nil {
+		return Meta{}, err
+	}
+
+	eofReader := NewEOFReader(data)
+
+	for !eofReader.isEOF() && !eofReader.hasError() {
+		var encKey [32]byte
+		_, err = rand.Read(encKey[:])
+		if err != nil {
+			return Meta{}, err
+		}
+
+		var encrypter eestream.Transformer
+
+		encrypter, err := eestream.NewEncrypter(&encKey, &nonce, s.encBlockSize, s.encType)
+		if err != nil {
+			return Meta{}, err
+		}
+
+		d := new([32]byte)
+		copy((*d)[:], (*derivedKey)[:])
+		encryptedEncKey, err := eestream.Encrypt(encKey[:], d, &nonce, s.encType)
+		if err != nil {
+			return Meta{}, err
+		}
 
-	for !awareLimitReader.isEOF() && !awareLimitReader.hasError() {
+		sizeReader := NewSizeReader(eofReader)
 		segmentPath := path.Prepend(fmt.Sprintf("s%d", totalSegments))
-		segmentData := io.LimitReader(awareLimitReader, s.segmentSize)
+		segmentReader := io.LimitReader(sizeReader, s.segmentSize)
+		peekReader := segments.NewPeekThresholdReader(segmentReader)
+		isStreamEncrypted, err := peekReader.IsLargerThan(encrypter.InBlockSize())
```

Comment:
I'm not following what's happening here with the isStreamEncrypted check. Why would checking the stream size determine that.

Reply:
The idea of the `peekReader.IsLargerThan(encrypter.InBlockSize())` check is to see if the data is larger than one encryption block. If larger then we are going to pad the data  and encrypt it with the eestream block transformer. Otherwise, we are encrypting it in one single call to the encrypt function. This is to avoid padding really small files.

For sure the `isStreamEncrypted` var needs a better name to reflect the above.

Reply:
Before it was called `isLarger` which was even worse I think. But I could change it back if yall think that's preferable.